src/back/wgsl/writer.rs

```diff
@@ -1353,9 +1353,9 @@ impl<W: Write> Writer<W> {
                             back::vector_size_str(rows)
                         )?;
                     }
-                    TypeInner::Vector { size, .. } => {
+                    TypeInner::Vector { size, width, .. } => {
                         let vector_size_str = back::vector_size_str(size);
-                        let scalar_kind_str = scalar_kind_str(kind);
+                        let scalar_kind_str = scalar_kind_str(kind, convert.unwrap_or(width));
```

Comment:
What's supposed to happen to code like this
```rust
let a: f64 = f64(0.0);
let b: f32 = bitcast<f32>(a);
```
or
```rust
let a: f32 = f32(0.0);
let b: f64 = bitcast<f64>(a);
```
From what I could see the wgsl spec doesn't define what happens if the width of the types isn't equal. Naga doesn't even like this code and uses the width of the inner expression, so the code doesn't compile
```
[2021-11-07T15:07:09Z ERROR naga::front::wgsl] Given type Scalar { kind: Float, width: 4 } doesn't match expected Scalar { kind: Float, width: 8 }
error: the type of `b` is expected to be [3]
  ┌─ wgsl:4:9
  │
4 │     let b: f32 = bitcast<f32>(a);
  │         ^ definition of `b`

Could not parse WGSL
```
I may have skipped something while reading the spec, so if @kvark could help us here

Reply:
The WGSL spec [only allows](https://gpuweb.github.io/gpuweb/wgsl/#bitcast-expr) conversions between values that are the same size. So the programs above should be rejected as type errors.